src/loaders/workspace_uploader/workspace_uploader.py

```diff
@@ -0,0 +1,551 @@
+            docker_host = docker_host[5:]
+
+        vol[job_dir] = {"bind": job_dir, "mode": "rw"}
+        vol[docker_host] = {"bind": "/run/docker.sock", "mode": "rw"}
+
+        return env, vol
+
+    def start_callback_server(
+        self, client, container_name, job_dir, kb_base_url, token, port
+    ):
+        env, vol = self.setup_callback_server_envs(job_dir, kb_base_url, token, port)
+        self.container = client.containers.run(
+            name=container_name,
+            image=loader_common_names.CALLBACK_UPLOADER_IMAGE_NAME,
+            detach=True,
+            network_mode="host",
+            environment=env,
+            volumes=vol,
+        )
+        time.sleep(2)
+
+    def stop_callback_server(self):
+        self.container.stop()
+        self.container.remove()
+
+
+def _get_parser():
+    parser = argparse.ArgumentParser(
+        description="PROTOTYPE - Upload assembly files to the workspace service (WSS).",
+        formatter_class=loader_helper.ExplicitDefaultsHelpFormatter,
+    )
+
+    required = parser.add_argument_group("required named arguments")
+    optional = parser.add_argument_group("optional arguments")
+
+    # Required flag argument
+    required.add_argument(
+        "--workspace_id",
+        required=True,
+        type=int,
+        help="Target workspace addressed by the permanent ID",
+    )
+    required.add_argument(
+        f"--{loader_common_names.KBASE_COLLECTION_ARG_NAME}",
+        type=str,
+        help="Create a collection and link in data to that collection from the overall workspace source data dir",
+    )
+    required.add_argument(
+        f"--{loader_common_names.SOURCE_VER_ARG_NAME}",
+        type=str,
+        help="Create a source version and link in data to that collection from the overall workspace source data dir",
+    )
+
+    # Optional argument
+    optional.add_argument(
+        "--root_dir",
+        type=str,
+        default=loader_common_names.ROOT_DIR,
+        help="Root directory.",
+    )
+    optional.add_argument(
+        "--token_filepath",
+        type=str,
+        help="A file path that stores KBase token",
+    )
+    optional.add_argument(
+        "--env",
+        type=str,
+        choices=loader_common_names.KB_ENV,
+        default="PROD",
+        help="KBase environment, defaulting to PROD",
+    )
+    optional.add_argument(
+        "--upload_file_ext",
+        type=str,
+        nargs="+",
+        default=UPLOAD_FILE_EXT,
+        help="Upload only files that match given extensions",
+    )
+    optional.add_argument(
+        "--keep_job_dir",
+        action="store_true",
+        help="Keep SDK job directory after upload task is completed",
+    )
+    return parser
+
+
+def _get_yaml_file_path(root_dir: str, assembly_dir_name: str) -> str:
+    """
+    Get the uploaded.yaml file path from NCBI/sourcedata directory.
+    """
+    file_path = os.path.join(get_work_dir(root_dir), assembly_dir_name, UPLOADED_YAML)
```

Comment:
This uses the `sourcedir` path and hardcodes NCBI as the source. It should use the same directory as the assembly file path found in `_fetch_assemblies_to_uplaod` (or whatever the calling function is) which in that case is `collectionssource`

Reply:
Probably just best to not have this method and have the calling method figure out the file path, since it could be different for different collections and source vers

Reply:
Updated!

Reply:
:+1: